test/CMakeLists.txt

```diff
@@ -40,6 +40,25 @@ macro(onedpl_add_test test_source_file)
         target_compile_options(${_test_name} PRIVATE $<$<CONFIG:Release>:-fno-strict-aliasing>)
     endif()
 
+    # Avoid speculative compilation of kernels which may not be supported by a device
+    # This workaround is relevant until the following feature is implemented:
+    # https://github.com/intel/llvm/blob/sycl/sycl/doc/design/OptionalDeviceFeatures.md
+    if (ONEDPL_BACKEND MATCHES "^(dpcpp|dpcpp_only)$")
+        if (CMAKE_CXX_COMPILER_ID STREQUAL IntelLLVM)
+            # Do not use the workaround unintentionally with newer versions if the feature is implmented
+            if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 2023.0 OR CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 2023.0)
```

Comment:
Would not `if (NOT ... VERSION_GREATER ...)` be better for now? We can later change to VERSION_LESS once the "good" version is known.